tutorial.md

```diff
@@ -0,0 +1,1022 @@
+`ppHidden` is for the hidden workspaces that have windows on them, etc.
+The rest is just deciding on some pretty colours and formatting things
+just how we like it.
+
+If this is too much for you, you can also really just start with the
+blank
+
+``` haskell
+  myXmobarPP :: PP
+  myXmobarPP = def
+```
+
+then add something, reload xmonad, see how things change and whether you
+like them.  If not, remove that part and try something else.  If you do,
+try to understand how that particular piece of code works.  You'll have
+something approaching the above that you fully understand in no time!
+
+## Configuring Related Utilities
+
+So now you've got a status bar and xmonad.  We still need a few more
+things: a screensaver, a tray for our apps that have tray icons, a way
+to set our desktop background, and the like.
+
+For this, we will need a few pieces of software.
+
+``` shell
+  apt-get install trayer xscreensaver
+```
+
+If you want a network applet, something to set your desktop background,
+and a power-manager:
+
+``` shell
+  apt-get install nm-applet feh xfce4-power-manager
+```
+
+First, configure xscreensaver how you like it with the
+`xscreensaver-demo` command.  Now, we will set these things up in
+`~/.xinitrc` (we could also do most of this in xmonad's `startupHook`,
+but `~/.xinitrc` is perhaps more standard).  That file may wind up
+looking like this:
+
+``` shell
+  #!/bin/sh
+
+  [... default stuff that your distro may throw in here ...]
+
+  # Set up an icon tray
+  trayer --edge top --align right --SetDockType true --SetPartialStrut true \
+   --expand true --width 10 --transparent true --tint 0x5f5f5f --height 18 &
+
+  # Set the default X cursor to the usual pointer
+  xsetroot -cursor_name left_ptr
+
+  # Set a nice background
+  feh --bg-fill --no-fehbg ~/.wallpapers/haskell-red-noise.png
+
+  # Fire up screensaver
+  xscreensaver -no-splash &
+
+  # Power Management
+  xfce4-power-manager &
+
+  if [ -x /usr/bin/nm-applet ] ; then
+     nm-applet --sm-disable &
+  fi
+
+  exec xmonad
+```
+
+Notice the call to `trayer` above.  The options tell it to go on the top
+right, with a default width of 10% of the screen (to nicely match up
+with xmobar, which we set to a width of 90% of the screen).  We give it
+a color and a height.
+
+Then we fire up the rest of the programs that interest us.
+
+Finally, we start xmonad.
+
+<img alt="blank desktop" src="https://user-images.githubusercontent.com/50166980/111529498-84d49080-8762-11eb-9e81-c15dd844b0a9.png" width="660">
+
+Mission accomplished!
+
+Of course substitute the wallpaper for one of your own.  If you like the
+one used above, you can find it [here](https://i.imgur.com/9MQHuZx.png).
+
+## Final Touches
+
+There may be some programs that you don't want xmonad to tile.  The
+classic example is Gimp; it pops up all sorts of new windows all the
+time, and they work best at defined sizes.  It makes sense for xmonad to
+ignore these kinds of windows.
```

Comment:
Do we want to call this "ignoring", given that `doIgnore` is also a thing (and generally not what is intended)?

Reply:
Good point, ignoring does have a specific meaning and floating a window is definitly not it—especially when we can just say what we want to do with the window: make it float